tiledb/sm/c_api/tiledb.cc

```diff
@@ -3274,6 +3274,129 @@ int32_t tiledb_array_encryption_type(
   return TILEDB_OK;
 }
 
+int32_t tiledb_array_put_metadata(
```

Comment:
Do you want to invert the noun/verb here? `tiledb_array_metadata_{put,get}` etc

Reply:
Hmm I left it like that on purpose because these functions pertain to a `tiledb_array_t` object and not a `tiledb_array_metadata_t` object. I am up for discussing it though.